dashboard/app/views/levels/levelGroup.js

```diff
@@ -20,86 +20,135 @@ function initLevelGroup(
+    var showConfirmationDialog = "levelgroup-submit-" + completeString;
+
     return {
       "response": JSON.stringify(lastAttempt),
       "result": true,
       "errorType": null,
-      "submitted": window.appOptions.level.submittable || forceSubmittable
+      "submitted": window.appOptions.level.submittable || forceSubmittable,
+      "showConfirmationDialog": showConfirmationDialog
     };
   };
 
-  function nextPage() {
-    location.href = location.href.replace("/page/" + page, "/page/" + (page + 1));
+  // Called by gotoPage and checkForChanges to save current answers.
+  // Calls the completeFn function when transmission is complete.
+  function saveAnswers(completeFn) {
+    var results = window.getResult();
+    var response = results.response;
+    var result = results.result;
+    var errorType = results.errorType;
+    var submitted = appOptions.submitted;
+
+    window.dashboard.reporting.sendReport({
+      program: response,
+      fallbackResponse: fallbackResponse,
+      callback: callback,
+      app: app,
+      level: level,
+      result: result,
+      pass: result,
+      testResult: result ? 100 : 0,
+      submitted: submitted,
+      onComplete: completeFn
+    });
   }
 
-  $(".nextPageButton").click(function () {
+  // Called by gotoPage when it's ready to actually change the page.
+  function changePage(targetPage) {
+    var newLocation = window.location.href.replace("/page/" + currentPage, "/page/" + targetPage);
+    window.location.href = newLocation;
+  }
+
+  // Called by previous/next button handlers.
+  // Goes to a new page, first saving current answers if necessary.
+  function gotoPage(targetPage) {
     // Are we read-only?  This can be because we're a teacher OR because an answer
     // has been previously submitted.
     if (window.appOptions.readonlyWorkspace) {
-      nextPage();
+      changePage(targetPage);
     } else {
       // Submit what we have, and when that's done, go to the next page of the
-      // long assessment.
-
-      var results = window.getResult();
-      var response = results.response;
-      var result = results.result;
-      var errorType = results.errorType;
-      var submitted = appOptions.submitted;
-
-      window.dashboard.reporting.sendReport({
-        program: response,
-        fallbackResponse: fallbackResponse,
-        callback: callback,
-        app: app,
-        level: level,
-        result: result,
-        pass: result,
-        testResult: result ? 100 : 0,
-        submitted: submitted,
-        onComplete: function () {
-          nextPage();
-        }
+      // long assessment.  Cancel any pending throttled attempts at saving state.
+      this.throttledSaveAnswers.cancel();
+      saveAnswers(function () {
+        changePage(targetPage);
       });
     }
-  });
+  }
+
+  $(".nextPageButton").click($.proxy(function (event) {
+    gotoPage(currentPage+1);
+  }, this));
```

Comment:
Nit: I don't think `$.proxy` is needed here because the inner function doesn't use `this`.